data/scripts.js

```diff
@@ -103,7 +103,11 @@ exports.BattleScripts = {
 	 */
 	useMove: function (move, pokemon, target, sourceEffect, zMove) {
 		if (!sourceEffect && this.effect.id) sourceEffect = this.effect;
-		if (zMove || (sourceEffect && sourceEffect.isZ)) {
+		if (zMove && move.id === 'weatherball') {
+			this.singleEvent('ModifyMove', move, null, pokemon, target, move, move);
+			move = this.getZMoveCopy(move, pokemon);
+			if (move.type !== 'Normal') this.add('-message', `Breakneck Blitz turned into ${move.name} due to the weather!`);
```

Comment:
To interact correctly with my client PR Zarel/Pokemon-Showdown-Client#894 you need to set the sourceEffect to the move Weather Ball. (Or we can change to some other source effect if you think it's more relevant.)